File: swagger-client/spring/src/main/java/io/swagger/model/EmailAddressParam.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * A single email address parameter used as the body of &#x60;POST /mail/blocks/delete&#x60;.
 */
@Schema(description = "A single email address parameter used as the body of `POST /mail/blocks/delete`.")
@Validated
@NotUndefined



public class EmailAddressParam   {
  @JsonProperty("email")

  private String email = null;


  public EmailAddressParam email(String email) { 

    this.email = email;
    return this;
  }

  /**
   * The email address to delist from all block lists.
   * @return email
   **/
  
  @Schema(example = "user@domain.com", required = true, description = "The email address to delist from all block lists.")
  
  @NotNull
  public String getEmail() {  
    return email;
  }



  public void setEmail(String email) { 

    this.email = email;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    EmailAddressParam emailAddressParam = (EmailAddressParam) o;
    return Objects.equals(this.email, emailAddressParam.email);
  }

  @Override
  public int hashCode() {
    return Objects.hash(email);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class EmailAddressParam {\n");
    
    sb.append("    email: ").append(toIndentedString(email)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: swagger-client/dart/lib/model/email_address_param.dart
part of swagger.api;

class EmailAddressParam {
  /* The email address to delist from all block lists. */
  String email = null;

  EmailAddressParam();

  @override
  String toString() {
    return 'EmailAddressParam[email=$email, ]';
  }

  EmailAddressParam.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    email = json['email'];
  }

  Map<String, dynamic> toJson() {
    return {
      'email': email
     };
  }

  static List<EmailAddressParam> listFromJson(List<dynamic> json) {
    return json == null ? new List<EmailAddressParam>() : json.map((value) => new EmailAddressParam.fromJson(value)).toList();
  }

  static Map<String, EmailAddressParam> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, EmailAddressParam>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new EmailAddressParam.fromJson(value));
    }
    return map;
  }
}
